packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
+      projectPath = res.path.trim();
     }
-    await copySample(path.join(sampleDir, 'activity.ts'), path.join(targetDir, 'activities.ts'));
-    await copySample(path.join(sampleDir, 'workflow.ts'), path.join(targetDir, 'workflows', 'example.ts'));
-    await copySample(path.join(sampleDir, 'interface.ts'), path.join(targetDir, 'interfaces', 'workflows.ts'));
   }
-}
 
-function getTemplate(sampleName: string): Template {
-  switch (sampleName) {
-    case 'hello-world':
-      return new HelloWorld('default');
-    case 'hello-world-mtls':
-      return new HelloWorld('mtls');
+  if (!projectPath) {
+    console.log();
+    console.log('Please specify the project directory:');
+    console.log(`  ${chalk.cyan(program.name())} ${chalk.green('<project-directory>')}`);
+    console.log();
+    console.log('For example:');
+    console.log(`  ${chalk.cyan(program.name())} ${chalk.green('my-temporal-project')}`);
+    console.log();
+    console.log(`Run ${chalk.cyan(`${program.name()} --help`)} to see all options.`);
+    process.exit(1);
   }
-  throw new TypeError(`Invalid sample name ${sampleName}`);
-}
 
-async function createProject(projectPath: string, useYarn: boolean, temporalVersion: string, sample: string) {
-  const root = path.resolve(projectPath);
-  const src = path.resolve(root, 'src');
-  const name = path.basename(root);
-  await mkdir(root);
-  const packageJson = { ...packageJsonBase, name };
-  await writePrettyJson(path.join(root, 'package.json'), packageJson);
-  await mkdir(src);
-  await mkdir(path.join(src, 'interfaces'));
-  await mkdir(path.join(src, 'workflows'));
-  await writePrettyJson(path.join(root, 'tsconfig.json'), tsConfig);
-  const sampleDir = path.join(__dirname, '../samples');
-  const template = getTemplate(sample);
-  await template.copySources(sampleDir, src);
-  if (useYarn) {
-    await spawn('yarn', ['install'], { cwd: root, stdio: 'inherit' });
-    await spawn('yarn', ['add', `temporalio@${temporalVersion}`], { cwd: root, stdio: 'inherit' });
-  } else {
-    await spawn(npm, ['install'], { cwd: root, stdio: 'inherit' });
-    await spawn(npm, ['install', `temporalio@${temporalVersion}`], { cwd: root, stdio: 'inherit' });
+  const resolvedProjectPath = path.resolve(projectPath);
+  const projectName = path.basename(resolvedProjectPath);
+
+  const { valid, problems } = validateNpmName(projectName);
+  if (!valid) {
+    console.error(
+      `Could not create a project called ${chalk.red(`"${projectName}"`)} because of npm naming restrictions:`
+    );
+
+    problems?.forEach((p) => console.error(`    ${chalk.red.bold('*')} ${p}`));
+    process.exit(1);
   }
-}
 
-async function init() {
-  const { _: args, ...opts } = arg({
-    '--use-yarn': Boolean,
-    '--temporal-version': String,
-    '--sample': String,
-  });
-  if (args.length !== 1) {
-    throw new UsageError();
+  let example = program.example;
+
+  // `example` is true when --example is used by itself
```

Comment:
It's weird that this is even possible with `commander`.

Reply:
It's an old version of commander. I briefly tried latest, which didn't work, and I didn't know if it would be worth upgrading.

Reply:
I think it's worth upgrading now rather than later.
If you use angle brackets in the flag definition you should only get a valid `string | undefined` and not boolean: https://www.npmjs.com/package/commander